projects/core/components/primitive-calendar/primitive-calendar.component.ts

```diff
@@ -110,8 +107,12 @@ export class TuiPrimitiveCalendarComponent {
             return null;
         }
 
-        if (value instanceof TuiDay) {
-            return value.daySame(item) ? TuiRangeState.Single : null;
+        if (!(value instanceof TuiDayRange)) {
+            if (value instanceof TuiDay) {
```

Comment:
Let's not do nested ifs, just move nested one up, before this one.

Reply:
fixed